ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageCommitter.scala

```diff
@@ -223,34 +221,21 @@ final private[kvutils] class PackageCommitter(
   }
 
   private def decodePackages(
-      hashesAndArchives: Iterable[(String, RawArchive)]
+      rawArchives: Iterable[RawArchive]
   ): Either[String, Map[Ref.PackageId, Ast.Package]] =
     metrics.daml.kvutils.committer.packageUpload.decodeTimer.time { () =>
-      type Result = Either[List[String], Map[Ref.PackageId, Ast.Package]]
-      hashesAndArchives
-        .foldLeft[Result](Right(Map.empty)) { case (acc, (hash, rawArchive)) =>
-          try {
-            acc.map { result =>
-              val archive = ArchiveParser.assertFromByteString(rawArchive.byteString)
-              result + lf.archive.Decode.assertDecodeArchive(archive)
-            }
-          } catch {
-            case NonFatal(e) =>
-              Left(
-                s"Cannot decode archive $hash: ${e.getMessage}" :: acc.left.getOrElse(Nil)
```

Comment:
This code didn't work as expected because of using `acc.map`.

Reply:
I’m not quite following. What was the issue with that?

Reply:
Because we were using `acc.map` it wasn't decoding the next archives after a failure, as `Either` is right-biased. So, effectively, there was no need to concatenate the "`Left`s", as there was always just one. I put some obvious stuff, is it more clear now, @cocreature?